sdk/python/feast/repo_operations.py

```diff
@@ -156,6 +168,9 @@ def apply_total(repo_config: RepoConfig, repo_path: Path):
             repo_config.offline_store, data_source
         )
 
+        if isinstance(data_source, BigQuerySource) and not data_source.query:
+            assert_table_exists(data_source.table_ref)
```

Comment:
Let's get rid of the if statement and call `data_source.validate()` here.

Reply:
btw, isn't it more Pythonic to do `is_valid()`? https://docs.python.org/3/library/stdtypes.html?highlight=identifier#str.isidentifier

Reply:
Perhaps it is standard for `is_<conditional>()` to return True or False, but `validate()` is ok since it makes assertions and would result in an exception on false? I'm open to either.